src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -205,7 +205,7 @@ def __init__(
 
         if self.model.config.model_type == "whisper":
             self.type = "seq2seq_whisper"
-        elif self.model.__class__ in MODEL_FOR_SPEECH_SEQ_2_SEQ_MAPPING.values():
+        elif self.model.__class__.__name__ in MODEL_FOR_SPEECH_SEQ_2_SEQ_MAPPING_NAMES.values():
```

Comment:
@sgugger @ArthurZucker Am I correct that this breaks if an user registered a model with `AutoModelForSpeechSeq2Seq.register(AutoConfig, MyCustomModel)`?

Reply:
From the description of the PR I would think that no (`_extra_content` is what should have been kept).
The `self._model_mapping._model_mapping = self` adds a field to `MODEL_FOR_SPEECH_SEQ_2_SEQ_MAPPING_NAMES` but it's not taken into account when computing the `values()` if I am not mistaken

Reply:
No new key is added `MODEL_FOR_SPEECH_SEQ_2_SEQ_MAPPING_NAMES`, but you can check for its `_model_mapping` if it exists, as is done in the diff below in `pipelines/base.py`